### PR TITLE
chore(changelog): add changelog generation

### PR DESCRIPTION
**Pull request by d3viant0ne**
- add conventional-changelog with angular presets
- add commitizen adapter
- rewrite contributing Documentation
- add `generate-changelog` script to write the changelog on release

Relates to #275

### CODE REVIEW

**Comment by danez** [comment]
👍

File: scripts/release/generate-changelog.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+'use strict'
+const fs = require('fs')
```
Comment (abouthiroppy) [inline scripts/release/generate-changelog.js]: Do we need ` semicolon`?

Comment (d3viant0ne) [inline scripts/release/generate-changelog.js]: Need, no.
Should have, yes. I'll update it in a sec.

**Comment by abouthiroppy** [comment]
Sorry for my late reply.
I checked this pr.

LGTM. Thanks😁

File: scripts/release/generate-changelog.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  releaseCount: isForce ? 0 : 1
+};
+
+const getOutputStream = () => {
+  return fs.createWriteStream('CHANGELOG.md');
+};
+
+const stream = cl(config).on('error', function(err) {
+    console.error(`An error occurred while generating the changelog:  ${err}`);
+  }).pipe(!isForce && addStream(inStream) || getOutputStream());
```
Comment (abouthiroppy) [inline scripts/release/generate-changelog.js]: [nit] Could you modify indent?

**Comment by danez** [comment]
We don't have that many releases so I think it is not really necessary to do this.

**Pull request closed without merging**